### PR TITLE
Add light/lit command packages

### PR DESCRIPTION
**Pull request by mmitche**
This adds light command package generation to aspnetcore.
After build of a wix project, generate a light package based off of the inputs that are sent to light.exe/lit.exe.

### CODE REVIEW

File: src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj [hunk 1]
```diff
@@ -40,4 +40,23 @@
...
+       package instead. -->
+  <Target Name="CreateWixPackageDrop" AfterTargets="Build">
+    <CreateLitCommandPackageDrop
+      LitCommandWorkingDir="$(WixCommandObjDir)"
+      WixExtensions="@(WixExtension)"
+      Bf="true"
+      Out="$(InstallersOutputPath)$(OutputName).wixlib"
+      WixSrcFiles="@(CompileObjOutput);@(WixObject);@(WixLibProjects);@(WixLibrary)">
+      <Output TaskParameter="LitCommandPackageNameOutput" PropertyName="_LitCommandPackageNameOutput" />
+    </CreateLitCommandPackageDrop>
```
Comment (dougbu) [inline src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:54]:
```suggestion
```

File: src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj [hunk 2]
```diff
@@ -40,4 +40,23 @@
   <PropertyGroup>
     <PackageFileName>$(InternalInstallerBaseName)-$(PackageVersion)-win-$(Platform)$(TargetExt)</PackageFileName>
   </PropertyGroup>
+
+  <!-- Overwrite the wix package drop creation target to create a lit
+       package instead. -->
+  <Target Name="CreateWixPackageDrop" AfterTargets="Build">
```
Comment (dougbu) [inline src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:46]: Doesn't this project import src/Installers/Windows/Wix.targets through https://github.com/dotnet/aspnetcore/blob/release/5.0/src/Installers/Windows/Directory.Build.targets#L3❔ If yes, why do we need an override here❔

File: src/Installers/Windows/Wix.targets [hunk 3]
```diff
@@ -72,4 +84,27 @@
...
+      Out="$(InstallersOutputPath)$(PackageFileName)"
+      AdditionalBasePaths="$(MSBuildProjectDirectory)"
+      WixExtensions="@(WixExtension)"
+      Loc="@(EmbeddedResource)"
+      Sice="$(SuppressIces)"
+      WixProjectFile="$(MSBuildProjectFile)"
+      Fv="true"
+      WixSrcFiles="@(CompileObjOutput);@(WixObject);@(WixLibProjects);@(_ResolvedWixLibraryPaths)">
+      <Output TaskParameter="LightCommandPackageNameOutput" PropertyName="_LightCommandPackageNameOutput" />
+    </CreateLightCommandPackageDrop>
```
Comment (dougbu) [inline src/Installers/Windows/Wix.targets:101]:
```suggestion
```

**Review by dougbu — COMMENTED** [review]
At a high level, why is this change needed❔ We already produce `*.msi` installers e.g. https://github.com/dotnet/aspnetcore/blob/release/5.0/src/Installers/Windows/Wix.props#L10-L15 and the MSIs in BlobArtifacts in (last release/5.0 build) https://dev.azure.com/dnceng/internal/_build/results?buildId=789717&view=artifacts&type=publishedArtifacts

File: src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj (hunk 2 above)
Comment (mmitche) [reply src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:46]: This is a different form of package drop (Lit lib creator vs. Light compiler). Most of the projects create installers, while this project creates a wixlib

**Comment by mmitche** [comment]
> At a high level, why is this change needed❔ We already produce `*.msi` installers e.g. https://github.com/dotnet/aspnetcore/blob/release/5.0/src/Installers/Windows/Wix.props#L10-L15 and the MSIs in BlobArtifacts in (last release/5.0 build) https://dev.azure.com/dnceng/internal/_build/results?buildId=789717&view=artifacts&type=publishedArtifacts

This is so these installers can have files replaced and re-linked after having those files signed. You can decompile msis, exes, etc. to do this operation, but it is very hard because database needs to be altered heavily after the files are signed (e.g. the length of files change). This change effectively just captures the pre-link artifacts (files mentioned in the wixobj's) and creates a little file drop and command line script that can be used the create the MSI. Then it becomes trivial to sign the input files and run the script to produce a signed msi.

**Comment by dougbu** [comment]
We sign the input files in a separate step from one one that builds the `*.wixproj` projects. But, this is about enabling the whole signing (Chrome was kind and auto-corrected that to "singing", must know this is probably good news 🎆) process to be moved out of our pipeline, right❔

File: src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj (hunk 1 above)
Comment (dougbu) [reply src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:54]: Ignore me on this. Missed the `<Output />` element

**Comment by mmitche** [comment]
> We sign the input files in a separate step from one one that builds the `*.wixproj` projects. But, this is about enabling the whole signing (Chrome was kind and auto-corrected that to "singing", must know this is probably good news 🎆) process to be moved out of our pipeline, right❔

Yep. This effectively creates a way to recreate the msis/exes/bundles/wixlib's out of the context of the build.

File: src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj (hunk 1 above)
Comment (mmitche) [reply src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:54]: Doesn't this create invalid xml given the Output element above?

Comment (dougbu) [reply src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:54]: Yup, that's why I said "ignore me" and resolved this point

Comment (mmitche) [reply src/Installers/Windows/SharedFrameworkLib/SharedFrameworkLib.wixproj:54]: > Yup, that's why I said "ignore me" and resolved this point

I think we were clicking the buttons and typing at the same time.

**Comment by mmitche** [comment]
@dougbu Depends on https://github.com/dotnet/aspnetcore/pull/25373

**Review by dougbu — APPROVED** [review]
So this will fail until #25373 goes in❔ Changes on top of that look fine though please remember to align the Microsoft.DotNet.Build.Tasks.Installers version and SHA when resolving conflicts with that PR (or whatever the final commit is if the current build fails).

**Comment by mmitche** [comment]
> So this will fail until #25373 goes in❔ Changes on top of that look fine though please remember to align the Microsoft.DotNet.Build.Tasks.Installers version and SHA when resolving conflicts with that PR (or whatever the final commit is if the current build fails).

Yep and will do.

**Review by chcosta — APPROVED** [review]

**Pull request merged**